lib/inventory_refresh/inventory_collection.rb

```diff
@@ -564,7 +579,9 @@ def db_collection_for_comparison_for(references)
     # @return [ActiveRecord::Relation] relation that can fetch all the references from the DB
     def full_collection_for_comparison
       return arel unless arel.nil?
-      parent.send(association)
+      rel = parent.send(association)
+      rel = rel.active if rel && supports_column?(:archived_on) && retention_strategy == :archive
```

Comment:
`supports_column?(:archived_on)` Looks like another place where we should have the archived_timestamp_column on the IC

Reply:
I also wonder if we shouldn't be calling `.active` on a relation since that assumes a lot as a gem.  I'd prefer the association we pass in only include the actives (and have e.g. all_vms/all_orch_stacks which includes the inactives) in the model

Reply:
right, I am not sure what is better. If we'd do it explicitly, we'd need to define each relation twice in the Source (and for through methods that would spread to every model?)

Reply:
Should we only do this if there's a `last_seen_on` column?

Reply:
Do do you want me to remove this, or lets keeps this until we do the scopes? (right now, this might spike memory for subcollections. For now we can observe the spike with mock collector)